### PR TITLE
Added option to filter slices by number of columns

### PR DESCRIPTION
**Pull request by luisenp**
Allows the user to filter the dataset so that it only returns slices of the desired number of columns.

### CODE REVIEW

File: fastmri/data/mri_data.py [hunk 1]
```diff
@@ -194,6 +197,10 @@ def __init__(
             num_examples = round(len(self.examples) * sample_rate)
             self.examples = self.examples[:num_examples]
 
+        if num_cols:
+            self.examples = [
+                ex for ex in self.examples if ex[2]["encoding_size"][1] in num_cols]
```
Comment (mmuckley) [inline fastmri/data/mri_data.py:202]: Nit: Could you format this with `black`?

File: fastmri/data/mri_data.py [hunk 2]
```diff
@@ -125,6 +127,7 @@ def __init__(
         challenge,
         sample_rate=1,
         dataset_cache_file=pathlib.Path("dataset_cache.pkl"),
+        num_cols=None
```
Comment (mmuckley) [inline fastmri/data/mri_data.py:130]: This needs to be added to `CombinedSliceDataset` as well.

**Review by mmuckley — CHANGES_REQUESTED** [review]
Just a couple of small requests. It would be good to add this feature to `CombinedSliceDataset` as well.

File: fastmri/data/mri_data.py (hunk 1 above)
Comment (luisenp) [reply fastmri/data/mri_data.py]: Ran `black` on the latest commit.

File: fastmri/data/mri_data.py (hunk 2 above)
Comment (luisenp) [reply fastmri/data/mri_data.py]: Added, can you check it makes sense?

**Review by mmuckley — APPROVED** [review]
LGTM

**Pull request merged**